wgpu/src/lib.rs

```diff
@@ -2522,6 +2547,62 @@ impl Texture {
             aspect: TextureAspect::All,
         }
     }
+
+    /// Returns the width of this `Texture`.
+    ///
+    /// This is always equal to the `size.width` that was specified when creating the texture.
+    pub fn width(&self) -> u32 {
+        self.width
+    }
+
+    /// Returns the height of this `Texture`.
+    ///
+    /// This is always equal to the `size.height` that was specified when creating the texture.
+    pub fn height(&self) -> u32 {
+        self.height
+    }
+
+    /// Returns the depth or layer count of this `Texture`.
+    ///
+    /// This is always equal to the `size.depth_or_array_layers` that was specified when creating the texture.
+    pub fn depth_or_array_layers(&self) -> u32 {
+        self.depth_or_array_layers
+    }
```

Comment:
Could we also add a "size" function that returns an Extent3D

Reply:
Sure thing. 👍